server/src/main/java/org/elasticsearch/common/network/InetAddresses.java

```diff
@@ -417,4 +417,29 @@ public static Tuple<InetAddress, Integer> parseCidr(String maskedAddress) {
     public static String toCidrString(InetAddress address, int prefixLength) {
         return new StringBuilder().append(toAddrString(address)).append("/").append(prefixLength).toString();
     }
+
+    /**
+     * Parse an IP address and its prefix length using the CIDR notation
+     * into a range of ip addresses corresponding to it.
+     * @throws IllegalArgumentException if the string is not formatted as {@code ip_address/prefix_length}
+     * @throws IllegalArgumentException if the IP address is an IPv6-mapped ipv4 address
+     * @throws IllegalArgumentException if the prefix length is not in 0-32 for IPv4 addresses and 0-128 for IPv6 addresses
+     * @throws NumberFormatException if the prefix length is not an integer
+     */
+    public static Tuple<InetAddress, InetAddress> parseIpRangeFromCidr(String maskedAddress) {
+        final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(maskedAddress);
```

Comment:
Would it be possible using something like a `record` instead of a tuple. A record allows us to name things like `lowerBound` and `upperBound` instead of having to use `v1/v2` and, in my opinion, make the code more readable.